doc/HttpLuaModule.wiki

```diff
@@ -5581,6 +5581,73 @@ According to the current implementation, only the parent coroutine (or "light th
 
 This API was first enabled in the <code>v0.9.9</code> release.
 
+== ngx.semaphore.new ==
+'''syntax:''' ''sem, err = ngx.semaphore.new(n)''
+
+'''context:''' ''init_worker_by_lua*, set_by_lua*, rewrite_by_lua*, access_by_lua*, content_by_lua*, header_filter_by_lua*, body_filter_by_lua, log_by_lua*, ngx.timer.*''
+
+Create a semaphore that has n resource. LuaJIT's ffi is need by this api.
+
+<geshi lang="lua">
+    local semaphore = require "ngx.semaphore"
+    local print = ngx.print
+    local sem,err = semaphore.new(0)
+    if not sem then
+        print("create semaphore failed: "..err)
+    end
+</geshi>
+
+== ngx.semaphore.post ==
+'''syntax:''' ''ok, err = sem:post()''
+
+'''context:''' ''init_worker_by_lua*, set_by_lua*, rewrite_by_lua*, access_by_lua*, content_by_lua*, header_filter_by_lua*, body_filter_by_lua, log_by_lua*, ngx.timer.*''
+
+The param sem is create by [[#ngx.semaphore.new|ngx.semaphore.new]]. Release one resource to a semaphore. If one light thread or coroutine is waiting on this semaphore, then it will be waked up. LuaJIT's ffi is need by this api.
+
+<geshi lang="lua">
+    local semaphore = require "ngx.semaphore"
+    local shared =  getmetatable(_G).__index
```

Comment:
Oh, man, this is bloodily hacky. We are against use of Lua globals especially such "true globals". Please see https://github.com/openresty/lua-nginx-module#lua-variable-scope

Instead, we should use proper Lua modules for worker-level data sharing in our own documenmtation, as documented here: https://github.com/openresty/lua-nginx-module#data-sharing-within-an-nginx-worker